code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java

```diff
@@ -275,7 +275,7 @@ void updateVolumeHostInfo(AllocationAttempt attempt, AllocationCandidate candida
             }
             if (VolumeConstants.ACCESS_MODE_SINGLE_HOST_RW.equals(v.getAccessMode())) {
                 persist = true;
-                v.setHostId(newHost);
+                DataAccessor.setField(v, VolumeConstants.FIELD_LAST_ALLOCATED_HOST_ID, newHost);
```

Comment:
setField is meant for api fields. We don't want this as an api field, we want it as a internal flag.
Instead use `DataAccessor.fromDataFieldOf(v).withKey(VolumeConstants.FIELD_LAST_ALLOCATED_HOST_ID).set(newHost)`